### ISSUE TITLE
[HOLD for payment 2023-07-24] [$2000] Settings page: remove code preventing rendering before `currentUserPersonalDetails` is ready

### ISSUE THREAD

**Issue by amyevans** [Expensify/App#18569]
Coming from https://github.com/Expensify/App/pull/17349#issuecomment-1536407713, 

We should refactor to remove the code that currently prevents the Settings screen from rendering before `currentUserPersonalDetails` is ready:
https://github.com/Expensify/App/blob/cdd8e319af4edbdb426fa3f78306c87b5a03c44a/src/pages/settings/InitialSettingsPage.js#L276-L282

A few notes:
* Nested screens can now be navigated to earlier than before
* Make sure there are no issues by (a) testing with a slow connection and (b) testing deep linking into the nested screens
* We _might_ also need some new loading states (e.g. after removing the above-mentioned code, the text below the avatar on the Settings screen flickers from email address to name + surname)

https://user-images.githubusercontent.com/7277067/236720288-95d05912-71e4-4e27-b051-3a723e922227.mov

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01157142bba66be571</li>
        <li>Upwork Job ID: 1655764572402282496</li>
        <li>Last Price Increase: 2023-05-18</li>
    </ul>
</details>

**Comment by amyevans** [comment]
@jczekalski and @aimane-chnaif, are you both willing to be assigned to this issue since you've got the context coming from #17349? Or should I open it up to other community members?

**Comment by VictoriyaSlonchak** [comment]
Hi, there

My name is Victoria Slonchak from Ukraine.

I have checked your requirement more closely.

To remove the code that currently prevents the Settings screen from rendering before currentUserPersonalDetails is ready, I think javasript try catch is very suitable. That is Refactoring CodeBase. 

try {
  // Dependent on currentUserPersonalDetails

} catch (error) {
  // handle the error
}

Thanks

**Comment by asadkhan-react** [comment]
Hi, I looked into this code very carefully. Can you show me an example of how the settings screen prevents rendering before currentUserPersonalDetails is ready?

**Comment by asadkhan-react** [comment]
Contributor details
Your Expensify account email: nazamfreelancer1234@gmail.com
Upwork Profile Link: https://www.upwork.com/freelancers/~01f5249775a5fb19ca

**Comment by VictoriyaSlonchak** [comment]
Contributor details
My Expensify account email: concierge@expensify.com
Upwork Profile Link: https://www.upwork.com/freelancers/~01d5b384030bd06c3e

**Comment by alitoshmatov** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Settings page: remove code preventing rendering before currentUserPersonalDetails is ready 
### What is the root cause of that problem?
Settings page is waiting for `currentUserPersonalDetails` to load before rendering its content
### What changes do you think we should make in order to solve the problem?
We should just remove these lines of code
https://github.com/Expensify/App/blob/cdd8e319af4edbdb426fa3f78306c87b5a03c44a/src/pages/settings/InitialSettingsPage.js#L280-L282
I have removed this line and it is working without any issues, you can see video below.

https://user-images.githubusercontent.com/59907218/237052983-6d11006e-7630-45ef-972b-b5f654a1cf48.mov

As said in description, the name is changing from email to display name(if display name exists), image also is changing, and workspace are appearing too. In faster network it might create flickering effect. 
If needed we can add loading state to track `currentUserPersonalDetails` status, and show skeletons for avatar and display name while loading is true. But I don't think it would be visually pleasing.Also, when request takes too much time it will just keep showing skeleton, which is worse than showing default avatar image, with email address.



### What alternative solutions did you explore? (Optional)

**Comment by idevcoder** [comment]
Contributor details
Your Expensify account email: developer.smartuv@gmail.com
Upwork Profile Link: https://www.upwork.com/freelancers/umangvaghasiya

**Comment by idevcoder** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Ensure that the appropriate UI is displayed before the profile details are ready to be shown in the settings menu.

### What is the root cause of that problem?
The current code is only returning null instead of any UI component until the currentUserPersonalDetails is ready.

### What changes do you think we should make in order to solve the problem?
Rather than returning null, we can include a loading placeholder that matches the user icon, name, role, and menu lists, similar to the box on the left side.

### What alternative solutions did you explore? (Optional)
None

**Comment by VictoriyaSlonchak** [comment]
Contributor details
My Expensify account email: royalsceptre88@gmail.com
Upwork Profile Link: https://www.upwork.com/freelancers/~01d5b384030bd06c3e

**Comment by stephanieelliott** [comment]
hey @aimane-chnaif looks like we have a few proposals here, can you review? 

https://github.com/Expensify/App/issues/18480#issuecomment-1537541205
https://github.com/Expensify/App/issues/18569#issuecomment-1539455780
https://github.com/Expensify/App/issues/18569#issuecomment-1539789013

**Comment by amyevans** [comment]
@jczekalski bump on the question - I think you were OOO at the end of last week. Otherwise we can add back `Help Wanted`

**Comment by stephanieelliott** [comment]
Moving back to `Help Wanted` so we can push this along.

**Comment by stephanieelliott** [comment]
Raised bounty to $2k

**Comment by alitoshmatov** [comment]
@aimane-chnaif Can you provide expected behavior?

**Comment by jczekalski** [comment]
@amyevans Yes, I was OOO at the end of last week and didn't get notified when you tagged me (not sure why). Can you assign the issue to me?

**Comment by amyevans** [comment]
@jczekalski how is this going? The expectation on `Daily` issues is at least 1 update per day, even if it is only clarifying that your time was spent elsewhere that day (with a link to the other GH ideally). Thanks!

**Comment by stephanieelliott** [comment]
Hi @jczekalski are you working on this? Please provide an update or link to a PR

**Comment by amyevans** [comment]
FYI: I'll be OOO tomorrow and Monday.

**Comment by jczekalski** [comment]
**Update**: Nothing new to report today. Work is currently in progress on [19234](https://github.com/Expensify/App/issues/19234) and [19236](https://github.com/Expensify/App/issues/19236). This is issue is next in line after proposals / PRs have been submitted for the other two issues.

**Comment by kowczarz** [comment]
I'm from Software Mansion, I'd like to work on this job.

**Comment by amyevans** [comment]
@aimane-chnaif bump to review the proposal please 😄

**Comment by kowczarz** [comment]
>entering these screens will cause a crash

@aimane-chnaif sorry, it was a typo.

Tomorrow morning I will edit proposal and elaborate more on how it will prevent flicker on Settings page.

**Comment by kowczarz** [comment]
@aimane-chnaif I've just updated the proposal

**Comment by sethraj14** [comment]
@aimane-chnaif Can you suggest the expected correct behaviour here? One option is to avoid displaying the profile sidebar when profile data is to present but seems like you have discarded this idea. Hence should we show the profile sidebar with a loader on the user fields until the data loads.

**Comment by shawnborton** [comment]
I personally don't mind this, but it would be nice if we made the display name fade in once it's loaded.

Alternatively, we could replace this with some kind of skeleton loader as well.

**Comment by shawnborton** [comment]
Or even just one. And then furthermore, it would be nice if that top Settings area had a fixed height so it doesn't jump around. I think @roryabraham is working on that in a separate issue.

**Comment by kowczarz** [comment]
Ok, so I will wait till @roryabraham will provide some info.

**Comment by stephanieelliott** [comment]
Hey @kowczarz and @aimane-chnaif - based on Rory's comments, does the proposal need to be updated? Or do we have everything we need here?

**Comment by kowczarz** [comment]
@aimane-chnaif does the skeletons work, or should I prepare some entering animations to the text components?

**Comment by kowczarz** [comment]
> This is also new feature. We should do that only when everyone agrees

Not sure what you mean by everyone. Is that everyone in the thread, or a few particular people.?

**Comment by amyevans** [comment]
I think showing nothing while loading the workspaces and payment balance is the best option. @shawnborton is OOO today but back tomorrow though, so let's wait a day for his input before finalizing and moving to PR.

**Comment by shawnborton** [comment]
I think showing nothing is okay as well, and then perhaps we just fade the information in once we have it.

**Comment by kowczarz** [comment]
@aimane-chnaif / @amyevans are we ready to start the PR?

**Comment by kowczarz** [comment]
BTW, I will be OOO tomorrow, on Monday and Tuesday, so I will be able to start coding on Wednesday.

**Comment by amyevans** [comment]
Yes, from my perspective we can move to PR, thanks!

**Comment by stephanieelliott** [comment]
Reapplying the `Bug` label to get another BZ member on this while I am OOO til July 3. Thanks @lschurr,  I'll grab this back from you when I return!

**Comment by lschurr** [comment]
Cool, thanks! @kowczarz any update?

**Comment by lschurr** [comment]
Any update @kowczarz - are you able to prioritize this?

**Comment by kowczarz** [comment]
Hey, sorry for not updating. I'm finishing 2 other tasks and I will handle this task. I think I should start implementing this issue tomorrow or a day after tomorrow.

**Comment by kowczarz** [comment]
Need some more time since [#21551](https://github.com/Expensify/App/pull/21551) is more complex then I expected.

**Comment by lschurr** [comment]
Any update @kowczarz?

**Comment by kowczarz** [comment]
Hi, I started work on it on Friday, but I'm also working on #21551 simultaneously since it requires waiting for builds.

**Comment by lschurr** [comment]
@stephanieelliott - I think you're back from OOO yeah? Going to unassign myself :)

**Comment by kowczarz** [comment]
PR is ready for review

**Comment by stephanieelliott** [comment]
PR is waiting on review from @amyevans

**Comment by stephanieelliott** [comment]
PR was deployed to staging

**Comment by stephanieelliott** [comment]
Per the calculations [here](https://github.com/Expensify/App/issues/18569#issuecomment-1634368834) there is no bonus for this one. 

  - External issue reporter: None, $0
 -  Contributor that fixed the issue: @kowczarz  - expert contrib., no issue payment 
 -  Contributor+ that helped on the issue and/or PR: @aimane-chnaif was paid in $2000 Upwork
